src/Symfony/Bundle/WebServerBundle/DependencyInjection/WebServerExtension.php

```diff
@@ -54,4 +60,16 @@ private function getPublicDirectory(ContainerBuilder $container)
 
         return $kernelProjectDir.'/'.$publicDir;
     }
+
+    private function getPidFileDirectory(ContainerBuilder $container): string
+    {
+        $kernelCacheDir = $container->getParameter('kernel.cache_dir');
+        $environment = $container->getParameter('kernel.environment');
+
+        if (basename($kernelCacheDir) !== $environment) {
+            return $container->getParameter('kernel.project_dir');
+        }
+
+        return \dirname($container->getParameter('kernel.cache_dir'));
```

Comment:
If I read this correctly, this means that the PID will be removed whenever the cache is clearer, which is not something I would expect. Shouldn't we store the PID under `var/cache/` directly?

Reply:
`dirname()` does it

Reply:
The default value for the `levels` argument in `dirname()` is `1`. So `\dirname($container->getParameter('kernel.cache_dir'));` will return the `var/cache` directory. Shall I add the `levels` argument to make it more obvious?

Reply:
sorry, I missed the `dirname()` here.